Source/MQTTnet.Extensions.ManagedClient/ManagedMqttClient.cs

```diff
@@ -94,24 +98,32 @@ public async Task StartAsync(IManagedMqttClientOptions options)
             _connectionCancellationToken = new CancellationTokenSource();
 
 #pragma warning disable CS4014 // Because this call is not awaited, execution of the current method continues before the call is completed
-            Task.Run(() => MaintainConnectionAsync(_connectionCancellationToken.Token), _connectionCancellationToken.Token);
+            _maintainConnectionTask = Task.Run(() => MaintainConnectionAsync(_connectionCancellationToken.Token), _connectionCancellationToken.Token);
 #pragma warning restore CS4014 // Because this call is not awaited, execution of the current method continues before the call is completed
 
             _logger.Info("Started");
         }
 
-        public Task StopAsync()
+        public async Task StopAsync()
         {
+            ThrowIfDisposed();
+
             StopPublishing();
             StopMaintainingConnection();
 
             _messageQueue.Clear();
 
-            return Task.FromResult(0);
+            if (_maintainConnectionTask != null)
+            {
+                await Task.WhenAny(_maintainConnectionTask);
+                _maintainConnectionTask = null;
```

Comment:
Why is here a "WhenAny" if there is only one? Is there a task missing in the parameter list?

Reply:
The big advantage of Task.WhenAny is that it only checks for completion but ignores exceptions within the task. If I use await _maintainConnectionTask it could cause an exception.